packages/stems/src/components/Button/Button.module.css

```diff
@@ -81,12 +82,24 @@
   transition: all var(--quick);
 }
 
+.button .overlay {
+  background-color: var(--overlay-color, transparent);
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  opacity: var(--overlay-opacity, 0);
```

Comment:
CSS variable inheritance is neat. If you tell an element with the `.overlay` class to consume a variable called `--overlay-opacity`, it will use the nearest value defined either on itself or its ancestors.

In the button hover/active classes below, we set values for this variable. And since those get applied to the parent `button` element, this element will inherit them and use it. 🎉